sklearn/ensemble/weight_boosting.py

```diff
@@ -765,12 +765,14 @@ def predict_proba(self, X):
                         for estimator, w in zip(self.estimators_,
                                                 self.estimator_weights_))
 
-        proba /= self.estimator_weights_.sum()
-        proba = np.exp((1. / (n_classes - 1)) * proba)
-        normalizer = proba.sum(axis=1)[:, np.newaxis]
-        normalizer[normalizer == 0.0] = 1.0
-        proba /= normalizer
-
+        if n_classes > 1:
```

Comment:
I think you should move the n_classes == 1 sooner to avoid to do non meaningful computation, e.g. something like:

```py
...
X = self._validate_X_predict(X)
if n_classes == 1:
    return np.ones((X.shape[0], 1))
... # the rest of the code can stay exactly the same without an else clause
```